scripts/language_model/transformer/softmax.py

```diff
@@ -0,0 +1,260 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Attention cells."""
+
+__all__ = ['AdaptiveLogSoftmaxWithLoss']
+
+import typing
+
+import mxnet as mx
+
+
+class AdaptiveLogSoftmaxWithLoss(mx.gluon.HybridBlock):
+    """Efficient softmax approximation
+
+    Grave, E., Joulin, A., Cissé, M., Jégou, H., & others, (2017). Efficient
+    softmax approximation for GPUs. In , Proceedings of the 34th International
+    Conference on Machine Learning-Volume 70 (pp. 1302–1310).
+
+    Parameters
+    ----------
+    vocab_size
```

Comment:
document type, too
vocab_size : int

Reply:
Type will be documented automatically (https://github.com/dmlc/gluon-nlp/pull/830)